File: 70.Climbing-Stairs.go
// https://leetcode.com/problems/climbing-stairs/
//
// algorithms
// Easy (42.68%)
// Total Accepted:    326,903
// Total Submissions: 765,977
// beats 100.0% of golang submission

package leetcode

func climbStairs(n int) int {
	if n <= 2 {
		return n
	}

	dp := make([]int, n+1)
	dp[1] = 1
	dp[2] = 2

	for i := 3; i <= n; i++ {
		dp[i] = dp[i-1] + dp[i-2]
	}

	return dp[n]
}

File: 70.Climbing-Stairs.java
// https://leetcode.com/problems/climbing-stairs/
//
// algorithms
// Easy (44.18%)
// Total Accepted: 403,545
// Total Submissions: 913,502

class Solution {
    public int climbStairs(int n) {
        if (n == 1) {
            return 1;
        }

        int st = 1, nd = 1;
        for (int i = 2; i <= n; i++) {
            int tmp = st + nd;
            st = nd;
            nd = tmp;
        }

        return nd;
    }
}